anvil/anvil-compiler/src/main/java/com/duckduckgo/anvil/compiler/ContributeToActivityStarterCodeGenerator.kt

```diff
@@ -111,6 +122,39 @@ class ContributeToActivityStarterCodeGenerator : CodeGenerator {
                     )
                     .build(),
             )
+            .addFunction(
+                FunSpec.builder("map")
+                    .addModifiers(KModifier.OVERRIDE)
+                    .addParameter(
+                        "deeplinkActivityParams",
+                        deeplinkActivityParamsFqName.asClassName(module),
+                    )
+                    .returns(
+                        activityParamsFqName.asClassName(module).copy(nullable = true),
+                    )
+                    .addCode(
+                        """
+                            val screenName = deeplinkActivityParams.screenName.takeUnless { it.isEmpty() } ?: return null
+                            return if (screenName == %S) {
```

Comment:
When screen name is not defined in `ContributeToActivityStarter`. This code generates something like
```kotlin
val screenName = ...
return if (screenName == "") {...} else {...}
```
The method eventually returns `null` because `val screenName = ...` returns null right in there, but still the following if statement is wrong.

I added a previous comment to skip the mapper generation when screen name is not defined in `ContributeToActivityStarter`, but anyways we should also guard against that here to make sure changes in codegen are independent.

```suggestion
                            val definedScreenName = %S
                            if (definedScreenName.isNullOrBlank()) return null
                            return if (screenName == definedScreenName) {
```